library/std/src/sys/unix/os.rs

```diff
@@ -493,17 +494,16 @@ pub unsafe fn environ() -> *mut *const *const c_char {
     &mut environ
 }
 
-pub unsafe fn env_lock() -> StaticMutexGuard {
-    // It is UB to attempt to acquire this mutex reentrantly!
-    static ENV_LOCK: StaticMutex = StaticMutex::new();
-    ENV_LOCK.lock()
+pub unsafe fn env_rwlock(readonly: bool) -> RWLockGuard {
+    static ENV_LOCK: RWLock = RWLock::new();
+    if readonly { ENV_LOCK.read_with_guard() } else { ENV_LOCK.write_with_guard() }
 }
 
 /// Returns a vector of (variable, value) byte-vector pairs for all the
 /// environment variables of the current process.
 pub fn env() -> Env {
     unsafe {
-        let _guard = env_lock();
+        let _guard = env_rwlock(true);
```

Comment:
It is not clear from reading this in isolation if the `true` means readonly or read-write

Reply:
Works on my machine 😛

![Screenshot_2021-02-08_10-48-34](https://user-images.githubusercontent.com/1065730/107203207-6fe63e00-69fb-11eb-9e06-0d15cbe16e1b.png)

But sure, I can make the enum public instead.

Reply:
I'd prefer a separate function for read and write locking. There's not really much benefit to keeping it as one function.